tensorflow/compiler/mlir/xla/transforms/hlo_legalize_to_lhlo.cc

```diff
@@ -265,30 +256,131 @@ class HloToLhloReturnConverter : public OpConversionPattern<xla_hlo::ReturnOp> {
 //   return
 //  }
 // }
-struct HloLegalizeToLhlo : public FunctionPass<HloLegalizeToLhlo> {
-  void runOnFunction() override {
+struct HloLegalizeToLhlo : public ModulePass<HloLegalizeToLhlo> {
+  void runOnModule() override {
     OwningRewritePatternList patterns;
-    ConversionTarget target(getContext());
+    auto& context = getContext();
+    ConversionTarget target(context);
     target.addLegalDialect<xla_lhlo::XlaLhloDialect>();
+    target.addLegalDialect<StandardOpsDialect>();
+    target.addIllegalOp<mlir::ReturnOp>();
+    target.addIllegalOp<mlir::TensorLoadOp>();
+    target.addIllegalOp<mlir::TensorStoreOp>();
+    target.addLegalOp<ModuleTerminatorOp>();
+    target.addIllegalDialect<xla_hlo::XlaHloDialect>();
+    target.addDynamicallyLegalOp<FuncOp>([&](FuncOp op) {
+      bool result = true;
+      for (auto& input : op.getType().getInputs()) {
+        result &= input.isa<MemRefType>();
+      }
+      return result;
+    });
+
+    auto module = getModule();
+    populateHLOToLHLOConversionPattern(module.getContext(), &patterns);
 
-    auto func = getFunction();
-    populateHLOToLHLOConversionPattern(func.getContext(), &patterns);
-    if (failed(applyPartialConversion(func, target, patterns, nullptr))) {
+    SmallVector<Operation*, 16> ops;
+    for (auto& op : module.getBody()->getOperations()) {
+      ops.push_back(&op);
+    }
+    if (failed(applyFullConversion(ops, target, patterns, nullptr))) {
```

Comment:
What error do you get with applyFullConversion(module.getBody()->getOperations(), ... ?

Reply:
```
C++ compilation of rule '//tensorflow/compiler/mlir/xla:hlo_legalize_to_lhlo' failed (Exit 1)
tensorflow/compiler/mlir/xla/transforms/hlo_legalize_to_lhlo.cc: In member function 'virtual void mlir::xla_hlo::{anonymous}::HloLegalizeToLhlo::runOnModule()':
tensorflow/compiler/mlir/xla/transforms/hlo_legalize_to_lhlo.cc:277:96: error: no matching function for call to 'applyFullConversion(mlir::Block::OpListType&, mlir::ConversionTarget&, mlir::OwningRewritePatternList&, std::nullptr_t)'
     if (failed(applyFullConversion(module.getBody()->getOperations(), target, patterns, nullptr))) {
                                                                                                ^
In file included from tensorflow/compiler/mlir/xla/transforms/hlo_legalize_to_lhlo.cc:30:0:
external/llvm-project/mlir/include/mlir/Transforms/DialectConversion.h:647:1: note: candidate: mlir::LogicalResult mlir::applyFullConversion(llvm::ArrayRef<mlir::Operation*>, mlir::ConversionTarget&, const mlir::OwningRewritePatternList&, mlir::TypeConverter*)
 applyFullConversion(ArrayRef<Operation *> ops, ConversionTarget &target,
 ^~~~~~~~~~~~~~~~~~~
external/llvm-project/mlir/include/mlir/Transforms/DialectConversion.h:647:1: note:   no known conversion for argument 1 from 'mlir::Block::OpListType {aka llvm::iplist<mlir::Operation>}' to 'llvm::ArrayRef<mlir::Operation*>'
external/llvm-project/mlir/include/mlir/Transforms/DialectConversion.h:651:1: note: candidate: mlir::LogicalResult mlir::applyFullConversion(mlir::Operation*, mlir::ConversionTarget&, const mlir::OwningRewritePatternList&, mlir::TypeConverter*)
 applyFullConversion(Operation *op, ConversionTarget &target,
 ^~~~~~~~~~~~~~~~~~~
external/llvm-project/mlir/include/mlir/Transforms/DialectConversion.h:651:1: note:   no known conversion for argument 1 from 'mlir::Block::OpListType {aka llvm::iplist<mlir::Operation>}' to 'mlir::Operation*'
Target //tensorflow/compiler/mlir:tf-opt failed to build
```
 Changed it to `applyFullConversion(module...)`.